sdk/python/lib/pulumi/output.py

```diff
@@ -232,7 +232,7 @@ def __getitem__(self, key: Any) -> 'Output[Any]':
         :return: An Output of this Output's underlying value, keyed with the given key as if it were a dictionary.
         :rtype: Output[Any]
         """
-        return self.apply(lambda v: UNKNOWN if isinstance(v, Unknown) else v[key], True)
+        return self.apply(lambda v: UNKNOWN if isinstance(v, Unknown) else cast(Any, v)[key], True)
```

Comment:
aside, is teh right way to express this something like `lambda v Union[Unknown, Any]`?  (is that something even expressible with python types?

Reply:
You can apply a type signature to a lambda if you assign it to a variable: https://github.com/pulumi/pulumi/blob/13c337fe344f9ccd76b5787e971ee41c987a66a8/sdk/python/lib/pulumi/output.py#L258